TESTS/netsocket/tls/main.cpp

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2019, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#define WIFI 2
+#if !defined(MBED_CONF_TARGET_NETWORK_DEFAULT_INTERFACE_TYPE) || \
+    (MBED_CONF_TARGET_NETWORK_DEFAULT_INTERFACE_TYPE == WIFI && !defined(MBED_CONF_NSAPI_DEFAULT_WIFI_SSID))
+#error [NOT_SUPPORTED] No network configuration found for this target.
+#endif
+#ifndef MBED_CONF_APP_ECHO_SERVER_ADDR
+#error [NOT_SUPPORTED] Requires parameters from mbed_app.json
+#endif
+
+#include "mbed.h"
+#include "greentea-client/test_env.h"
+#include "unity/unity.h"
+#include "utest.h"
+#include "utest/utest_stack_trace.h"
+#include "tls_tests.h"
+
+#if defined(MBEDTLS_SSL_CLI_C) || defined(DOXYGEN_ONLY)
```

Comment:
Instead of this, can you do
```
#if !defined(MBEDTLS_SSL_CLI_C)
#error [NOT_SUPPORTED] This device does not support SSL library
#endif
```

This `[NOT SUPPORTED]` is magical (undocumented) macro that causes this testbinary to be skipped from build.

Reply:
Thank you for the hint. I still had to exclude the source code from building, otherwise I was getting more errors and I am afraid CI wouldn't accept that.
I replaced the empty main() with the suggested error macro. Locally it tells me that the build gets skipped:
```
Build successes:
  * ARCH_PRO::GCC_ARM::MBED-BUILD
Build skips:
  * ARCH_PRO::GCC_ARM::TESTS-NETSOCKET-TLS
```
@0xc0170 , will CI accept this as a non-failure or will it flag the build red?

Reply:
Yes,

Build skips: ... is the correct state.

CI does not try to run those, so it does not cause failures.